bin/jobserver

```diff
@@ -0,0 +1,386 @@
+
+    __job_free_slot_lock__ = threading.Semaphore(__max_jobs__)
+    __job_running_lock__ = threading.Semaphore(__max_jobs__)
+
+    if options.mem == 0:
+        __min_free_mem__ = None
+    else:
+        __min_free_mem__ = options.mem * 1024
+
+    if options.load == 0:
+        __max_load__ = None
+    else:
+        __max_load__ = options.load
+
+    if options.ppid == 0:
+        __parent_pid__ = None
+    else:
+        __parent_pid__ = options.ppid
+
+    # Check if PID file exists
+    if os.path.isfile(options.pid):
+        with open(options.pid, "r") as f:
+            try:
+                pid = int(f.read())
+            except ValueError:
+                pass
+            else:
+                try:
+                    os.kill(pid, 0)
+                    __primary_server__ = False
+                except PermissionError:
+                    __primary_server__ = False
+                except ProcessLookupError:
+                    pass
+
+    # Check if failback PID file exists
+    if (
+        __primary_server__
+        and options.pid == "/var/run/jobserver.pid"
+        and os.path.isfile("/tmp/jobserver.pid")
+    ):
+        with open("/tmp/jobserver.pid", "r") as f:
+            try:
+                pid = int(f.read())
+            except ValueError:
+                pass
+            else:
+                try:
+                    os.kill(pid, 0)
+                    __primary_server__ = False
+                except PermissionError:
+                    __primary_server__ = False
+                except ProcessLookupError:
+                    pass
+
+    if __primary_server__:
+        # Create PID file
+        try:
+            with open(options.pid, "w") as f:
+                f.write(str(os.getpid()))
+        except OSError:
+            if options.pid == "/var/run/jobserver.pid":
+                try:
+                    with open("/tmp/jobserver.pid", "w") as f:
+                        f.write(str(os.getpid()))
+                        options.pid = "/tmp/jobserver.pid"
+                except OSError:
+                    print("Can't write PID file")
+                    sys.exit(1)
+            else:
+                print("Can't write PID file")
+                sys.exit(1)
+    else:
+        # print("Job server is already running")
+        pass
+
+    if __primary_server__:
+        __show_status__ = options.stats
+    else:
+        __show_status__ = False
+
+    if __parent_pid__:
+        wait_parent_thread = threading.Thread(target=wait_parent, daemon=True)
+        wait_parent_thread.start()
+
+    status_thread = threading.Thread(target=job_status, daemon=True)
+    status_thread.start()
+
+    if __primary_server__:
+        ThreadingTCPServer.allow_reuse_address = True
+        __server__ = ThreadingTCPServer((str(__address__), options.port), JobServer)
+        try:
+            __server__.serve_forever()
+        except KeyboardInterrupt:
+            __server__.shutdown()
```

Comment:
It would be nice to print a newline (or do something similar) after `Ctrl-C` so that the shell prompt doesn't get printed on the same line as the jobserver status info when running `jobserver -s`